backends/bmv2/psa_switch/psaSwitch.h

```diff
@@ -47,7 +47,35 @@ class PsaSwitchExpressionConverter : public ExpressionConverter {
                                       ProgramStructure* structure, cstring scalarsName) :
     BMV2::ExpressionConverter(refMap, typeMap, structure, scalarsName) { }
 
+    /**
+     * Checks if a Parameter is of type PSA_CounterType_t returns true
+     * if it is, false otherwise.
+     */
+    bool isCounterMetaData(const IR::Parameter* param) {
+      return param->type->toString() == "PSA_CounterType_t";
+    }
+
     Util::IJson* convertParam(UNUSED const IR::Parameter* param, cstring fieldName) override {
+        if (isCounterMetaData(param)) {  // check if its counter metadata
+          auto jsn = new Util::JsonObject();
+          jsn->emplace("name", param->toString());
+          jsn->emplace("type", "hexstr");
+          auto bitwidth = param->type->width_bits();
+
+          // encode the counter type from enum -> int
+          if (fieldName == "BYTES") {
+            cstring repr = BMV2::stringRepr(0, ROUNDUP(bitwidth, 32));
+            jsn->emplace("value", repr);
+          } else if (fieldName == "PACKETS") {
+            cstring repr = BMV2::stringRepr(1, ROUNDUP(bitwidth, 32));
+            jsn->emplace("value", repr);
+          } else {
```

Comment:
you may want to validate the third case as well and give an error otherwise.

Reply:
Added an error reporting function as well, thanks